Add osx to travis

mariuz opened the pull request:
Add osx to travis also fix a few compile issues on osx (i used a few functions from posix/mod_loader.cpp)

mariuz committed:
Update .travis.yml

mariuz committed:
Update .travis.yml

mariuz committed:
Update .travis.yml

mariuz committed:
Update .travis.yml

mariuz committed:
Update .travis.yml

mariuz committed:
macosx doesn't have posixfadvise

mariuz committed:
add -msse4.1

mariuz committed:
msse4.2 is needed

mariuz committed:
add os_utils

mariuz committed:
Update mod_loader.cpp

mariuz committed:
Update mod_loader.cpp

mariuz committed:
use gnu sed 

otherwise we get the bellow error : undefined label

sed -n -e "s/.*btyacc: \(.*conflicts.*\)/\1/p" /Users/travis/build/mariuz/firebird/gen/y.txt > /Users/travis/build/mariuz/firebird/src/dsql/parse-conflicts.txt
sed -i 's/#define \([A-Z].*\)/#define TOK_\1/' /Users/travis/build/mariuz/firebird/gen/y_tab.h
sed: 1: "/Users/travis/build/mar ...": undefined label 'ravis/build/mariuz/firebird/gen/y_tab.h'

https://mpdaugherty.wordpress.com/2010/05/27/difference-with-sed-in-place-editing-on-mac-os-x-vs-linux/

mariuz committed:
verify sed path

mariuz committed:
Update .travis.yml

asfernandes commented on .travis.yml:
```diff
@@ -10,23 +16,55 @@ branches:
+      export CXX=clang++
+      export CC=clang
+      alias libtoolize=glibtoolize
+      git clone https://github.com/libtom/libtomcrypt.git
```
Why not use `--with-builtin-tomcrypt`?

mariuz replied:
I think there were some issues when building , i will recheck

mariuz committed:
brew install sed and add it to the path

mariuz committed:
no need to uninstall brew uninstall gnu-sed

add /usr/local/bin to the path

mariuz committed:
use INLINE_EDIT_SED

mariuz committed:
use osx sed with INLINE_EDIT_SED

mariuz committed:
do not install gnu sed

mapopa committed:
Merge remote-tracking branch 'upstream/master'

mariuz committed:
Update .travis.yml

mariuz committed:
Update .travis.yml

mariuz committed:
Update .travis.yml

mapopa committed:
use flags from 3.0.x branch
https://github.com/FirebirdSQL/firebird/commit/c165a1d980b6673d0df0121d0eb205f981944967

mariuz committed:
cleanup . no need for gnu sed

mariuz committed:
add also -std=c++11 to CXXFLAGS

mariuz committed:
test also DYLD_FALLBACK_LIBRARY_PATH

mariuz committed:
add also clang to linux builds

mariuz committed:
add the rest of compilers

mariuz committed:
Update .travis.yml

mariuz committed:
compiler matrix didn't worked as i expected

mariuz commented:
I will close this one and start another pull request
